app/src/main/java/ca/carleton/gcrc/n2android_mobile1/connection/ConnectionSyncProcess.java

```diff
@@ -188,56 +163,93 @@ public void updateLocalDocuments(List<JSONObject> documents) throws Exception {
 
-            Revision revisionRecord = trackingDb.getRevisionFromDocId(docId);
-            if( null == revisionRecord ){
-                revisionRecord = new Revision();
-            }
-            revisionRecord.setDocId(docId);
-
-            if(!remoteRev.equals(revisionRecord.getRemoteRevision())){
-                CouchbaseDocInfo info;
-                if( documentDb.documentExists(docId) ) {
-                    JSONObject existingDoc = documentDb.getDocument(docId);
-                    String existingRev = existingDoc.optString("_rev",null);
-                    if( null != existingRev ){
-                        doc.put("_rev",existingRev);
-                    }
-                    info = documentDb.updateDocument(doc);
-                } else {
-                    // When creating a document, no revision should be set
-                    doc.remove("_rev");
-
-                    info = documentDb.createDocument(doc);
-                }
+            You have no local changes
 
-                revisionRecord.setRemoteRevision(remoteRev);
-                revisionRecord.setLocalRevision(info.getRev());
-                trackingDb.updateRevision(revisionRecord);
+            - OR -
+
+            You have committed your local changes and you have no made changes since your previous
+            commit.
+         */
+        boolean isUnchangedLocalVersion = !changedLocally || (changedLocally && !isNewCommit);
+
+        if (changedRemote && isUnchangedLocalVersion) {
+            Revision revision = getRevisionRecord(docId);
+            updateDocumentDatabase(remoteDoc, revision);
+            return true;
+        }
 
-                updated = true;
+        if (changedLocally && isNewCommit) {
+            updateServerDocument(localDocument);
+            return true;
+        }
+
+        return false;
+    }
+
+    public Revision getRevisionRecord(String docId) throws Exception {
+        Revision revisionRecord = trackingDb.getRevisionFromDocId(docId);
+        if( null == revisionRecord ){
+            revisionRecord = new Revision();
+        }
+        revisionRecord.setDocId(docId);
+
+        return revisionRecord;
+    }
+
+    public void updateDocumentDatabase(JSONObject doc, Revision revisionRecord) throws Exception {
+        String docId = doc.getString("_id");
+        String remoteRev = doc.optString("_rev", null);
+
+        CouchbaseDocInfo info;
+        if( documentDb.documentExists(docId) ) {
+            JSONObject existingDoc = documentDb.getDocument(docId);
+            String existingRev = existingDoc.optString("_rev",null);
+            if( null != existingRev ){
+                doc.put("_rev",existingRev);
             }
+            info = documentDb.updateDocument(doc);
+        } else {
+            // When creating a document, no revision should be set
+            doc.remove("_rev");
 
-            return updated;
+            // TODO: Find a way to keep the attachments on the document.
```

Comment:
perhaps we should capture items like these in new cards or checklist items on your card, what do you think?